### PR TITLE
Solution to LeetCode problem: 287

### PR DESCRIPTION
**Pull request by Amisha328**
[Problem link: ](https://leetcode.com/problems/find-the-duplicate-number/)

### CODE REVIEW

**Comment by Amisha328** [comment]
Hey @Tahanima,
Please check PR #38 and let me know if any further changes are required.

File: Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Solution {
+public:
+    int findDuplicate(vector<int>& nums) {
+        sort(nums.begin(),nums.end());
```
Comment (Tahanima) [inline Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp:4]: Change this line to this:
```cpp
sort(nums.begin(), nums.end());
```

File: Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        int n =nums.size();
```
Comment (Tahanima) [inline Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp:5]: Change this line to this:
```cpp
int n = nums.size();
```

File: Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        for(int i = 0;i<n-1;i++)
```
Comment (Tahanima) [inline Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp:6]: Change this line to this:
```cpp
for (int i = 0; i < n-1; i++)
```

File: Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        {
+            if(nums[i] == nums[i+1])
+                return nums[i];
+        }
+        return -1;
+    }
+};
+
+// Time Complexity: O(n)
```
Comment (Tahanima) [inline Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp:15]: Your solution's time complexity is O(nlogn) because of calling the sort function.

**Review by Tahanima — CHANGES_REQUESTED** [review]
Hi @Amisha328 
Please resolve the requested changes.

File: Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp (hunk 4 above)
Comment (Amisha328) [reply Algorithms/Medium/287_FindTheDuplicateNumber/Solution.cpp:15]: Oh, yes, thank you. I will do make changes.

**Comment by Amisha328** [comment]
Thank You @Tahanima.
I will make all the suggested changes ASAP.

**Comment by Amisha328** [comment]
Hey, @Tahanima.
I have made the suggested changes. Please let me know if any further changes are required.

**Comment by Tahanima** [comment]
This resolves #17

**Review by Tahanima — APPROVED** [review]
Great work! :fire: Thank you so much for your contribution! :grinning: and welcome to **LeetCode Solution Curation** as a **Contributor**

Cheers! 🙌

**Comment by Amisha328** [comment]
> Great work! 🔥 Thank you so much for your contribution! 😀 and welcome to **LeetCode Solution Curation** as a **Contributor**
> 
> Cheers! 🙌

Thank you so much. Happy to contribute. 😀

**Pull request merged**